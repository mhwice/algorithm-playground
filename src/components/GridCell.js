import React from "react";
import PropTypes from "prop-types";

const GridCell = ({ isStart, isEnd }) => {
	let classModifierName;
	if (isStart) {
		classModifierName = "grid-cell-start";
	} else if (isEnd) {
		classModifierName = "grid-cell-end";
	} else {
		classModifierName = "";
	}

	// ! If possible, should refactor so that there is not a space after grid-cell for regular cells
	return <div className={`grid-cell ${classModifierName}`} />;
};

GridCell.defaultProps = {
	isStart: false,
	isEnd: false
};

GridCell.propTypes = {
	isStart: PropTypes.bool,
	isEnd: PropTypes.bool
};

export { GridCell as default };
