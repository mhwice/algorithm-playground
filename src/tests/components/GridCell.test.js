import React from "react";
import { shallow } from "enzyme";
import GridCell from "../../components/GridCell";

test("should render GridCell correctly", () => {
	const wrapper = shallow(<GridCell />);
	expect(wrapper).toMatchSnapshot();
});
