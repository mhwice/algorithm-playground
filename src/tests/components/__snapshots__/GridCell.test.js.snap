// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render GridCell correctly 1`] = `
<div
  className="grid-cell "
/>
`;
